server/middleware/checkPrisonerInCaseloadMiddleware.ts

```diff
@@ -0,0 +1,44 @@
+import { RequestHandler } from 'express'
+import ServerError from '../utils/serverError'
+import NotFoundError from '../utils/notFoundError'
+import { prisonerBelongsToUsersCaseLoad, userHasRoles } from '../utils/utils'
+import { Role } from '../data/enums/role'
+import { addMiddlewareError } from './middlewareHelpers'
+
+export default function checkPrisonerInCaseload({ allowGlobal = true, allowInactive = true } = {}): RequestHandler {
+  return async (req, res, next) => {
+    const prisonerData = req.middleware?.prisonerData
+
+    if (!prisonerData) {
+      return next(new ServerError('CheckPrisonerInCaseloadMiddleware: No PrisonerData found in middleware'))
+    }
+
+    const globalSearchUser = userHasRoles([Role.GlobalSearch], res.locals.user.userRoles)
+    const canViewInactiveBookings = userHasRoles([Role.InactiveBookings], res.locals.user.userRoles)
+    const inactiveBooking = ['OUT', 'TRN'].some(prisonId => prisonId === prisonerData.prisonId)
```

Comment:
Looks good. I think the inactiveBooking logic could be it's own middleware or util. Maybe a future iteration.